### ISSUE TITLE
NoSuchElementException in LintWorkerRunnable

### ISSUE THREAD

**Issue by REPLicated** [jeremymailen/kotlinter-gradle#101]
With 1.25.1, I got the following stacktrace. I cannot reliably reproduce it, which might hint at some concurrency issue.

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':server:core:lintKotlinMain'.
> There were multiple failures while executing work items
   > A failure occurred while executing org.jmailen.gradle.kotlinter.tasks.lint.LintWorkerRunnable
      > java.util.NoSuchElementException (no error message)
   > A failure occurred while executing org.jmailen.gradle.kotlinter.tasks.lint.LintWorkerRunnable
      > java.util.NoSuchElementException (no error message)
```

**Comment by jeremymailen** [comment]
Thank you for the report! If you are able to capture the `gradle --stacktrace` next time it happens that will help identify the problem.

Does it happen to occur and then go away between code edits? I had run into similar error messages before when I was running `lintKotlin` standalone rather than after a compile and when I got the full stacktrace it ended up being that my code had a syntax error and the exception was originating from the lint rules attempting to parse.

**Comment by REPLicated** [comment]
@jeremymailen : Unfortunately it has never occured when I supplied `--stacktrace`. I guesstimate the failure happens in <5% of the invocations.

However, I can confirm that the build failures are not related to local code edits. Our master branch CI pipeline has failed once with the same error yesterday. (If it helps: The builds run in Docker containers so the Gradle daemon is never reused).

```
16:10:48  Execution failed for task ':server:core:lintKotlinMain'.
16:10:48  > There were multiple failures while executing work items
16:10:48     > A failure occurred while executing org.jmailen.gradle.kotlinter.tasks.lint.LintWorkerRunnable
16:10:48        > java.util.NoSuchElementException (no error message)
16:10:48     > A failure occurred while executing org.jmailen.gradle.kotlinter.tasks.lint.LintWorkerRunnable
16:10:48        > Index 2 out of bounds for length 2
```

**Comment by REPLicated** [comment]
@jeremymailen : Here is a stacktrace. Probably related to or caused by the fix of #100 

```
Caused by: java.util.NoSuchElementException
        at java.base/jdk.internal.loader.URLClassPath$1.nextElement(URLClassPath.java:356)
        at java.base/jdk.internal.loader.URLClassPath$1.nextElement(URLClassPath.java:331)
        at org.jmailen.gradle.kotlinter.support.RuleSetsKt.resolveRuleSets(ruleSets.kt:27)
        at org.jmailen.gradle.kotlinter.tasks.lint.LintWorkerRunnable.run(LintWorkerRunnable.kt:47)
        at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:39)
        at org.gradle.workers.internal.NoIsolationWorkerFactory$1$1.execute(NoIsolationWorkerFactory.java:61)
        at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:44)
        at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:41)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
```

**Comment by jeremymailen** [comment]
Ok, thanks for the stacktrace! I'll get a fix out soon. It seems that ServiceLoaders are lazy on iteration, so maybe we ought to greedily produce a collection of RuleSetProviders at the root of the plugin.

**Comment by jeremymailen** [comment]
Ok, @REPLicated it should be fixed in `1.25.2`. Please give it a try and report back!
